### PR TITLE
refactor(testing): add methods for accessing projected content in component container harnesses

### PR DESCRIPTION
**Pull request by crisbeto**
Switches over all the harnesses for components that can contain projected user content to extend `ContentContainerComponentHarness` which will allow consumers to query for harnesses inside the element.

I've also deprecated a few similar methods that were added before `ContentContainerComponentHarness` was available. That being said, I wasn't totally sure whether the testing APIs fall under the same deprecation policy as everything else so I played it safe.

Finally, I cleaned up some `@dynamic` annotations which weren't doing anything, as far as I could tell.

### CODE REVIEW

**Review by mmalerba — APPROVED** [review]
LGTM

**Review by devversion — APPROVED** [review]
LGTM. One minor comment (I reviewed this morning but forgot submitting..)

File: src/material-experimental/mdc-menu/testing/menu-harness.ts [hunk 1]
```diff
@@ -6,12 +6,20 @@
...
+  HarnessQuery,
+  TestElement,
+  TestKey,
+} from '@angular/cdk/testing';
 import {coerceBooleanProperty} from '@angular/cdk/coercion';
 import {MenuHarnessFilters, MenuItemHarnessFilters} from '@angular/material/menu/testing';
 
 /** Harness for interacting with an MDC-based mat-menu in tests. */
-export class MatMenuHarness extends ComponentHarness {
+export class MatMenuHarness extends ContentContainerComponentHarness<string> {
```
Comment (devversion) [inline src/material-experimental/mdc-menu/testing/menu-harness.ts:22]: Should we `ContentContainerComponentHarness` rather? (for all similar harnesses too). That would help us enforce the methods being properly overridden and a consistent signature/API. Also looks like the base class isn't used at runtime anyway.

Comment (crisbeto) [reply src/material-experimental/mdc-menu/testing/menu-harness.ts:22]: I'll do a follow-up to try and make it so we don't have to override these methods, but we pass in something through the constructor instead.

Comment (devversion) [reply src/material-experimental/mdc-menu/testing/menu-harness.ts:22]: Sounds great 👍

**Pull request merged**